test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,226 @@
+    using index_cfg = seqan::FastFMIndexConfig<void, uint32_t>;
+    using fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::FMIndex<void, index_cfg>>;
+    using bi_fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::BidirectionalIndex<seqan::FMIndex<void, index_cfg>>>;
+
+    seqan2_rng_t make_sequence_seqan2()
+    {
+        seqan2_rng_t result;
+
+        if constexpr(seqan3::dimension_v<rng_t> == 1)
+        {
+            seqan::resize(result, length);
+            std::transform(seqan3.begin(), seqan3.end(), seqan::begin(result), seqan3::to_char);
+        }
+        else
+        {
+            seqan::resize(result, collection_size);
+            for (size_t i = 0; i < collection_size; ++i)
+            {
+                seqan::resize(result[i], length);
+                std::transform(seqan3[i].begin(), seqan3[i].end(), seqan::begin(result[i]), seqan3::to_char);
+            }
+        }
+
+        return result;
+    }
+#endif // SEQAN3_HAS_SEQAN2
+
+public:
+    rng_t seqan3{make_sequence_seqan3()};
+
+    sequence_generator() = default;
+    sequence_generator(sequence_generator const &) = default;
+    sequence_generator(sequence_generator &&) = default;
+    sequence_generator & operator=(sequence_generator const &) = default;
+    sequence_generator & operator=(sequence_generator &&) = default;
+    ~sequence_generator() = default;
+    sequence_generator(size_t length_, size_t collection_size_) : length{length_}, collection_size(collection_size_) {};
+
+#if SEQAN3_HAS_SEQAN2
+    seqan2_rng_t seqan2{make_sequence_seqan2()};
+    using index_t = std::conditional_t<std::same_as<index_tag_t, fm_index_seqan2>,
+                                       fm_index_seqan2_t,
+                                       std::conditional_t<std::same_as<index_tag_t, bi_fm_index_seqan2>,
+                                                          bi_fm_index_seqan2_t,
+                                                          std::conditional_t<std::same_as<index_tag_t, fm_index_seqan3>,
+                                                                             fm_index_seqan3_t,
+                                                                             bi_fm_index_seqan3_t>>>;
+#else
+    using index_t = std::conditional_t<std::same_as<index_tag_t, fm_index_seqan3>,
+                                       fm_index_seqan3_t,
+                                       bi_fm_index_seqan3_t>;
+#endif
+};
+
+template <typename index_tag_t, typename rng_t>
+void index_benchmark(benchmark::State & state)
+{
+    sequence_generator<index_tag_t, rng_t> generator(state.range(0), state.range(1));
+    for (auto _ : state)
+    {
+#if SEQAN3_HAS_SEQAN2
+        if constexpr (std::same_as<index_tag_t, fm_index_seqan2> || std::same_as<index_tag_t, bi_fm_index_seqan2>)
+        {
+            typename sequence_generator<index_tag_t, rng_t>::index_t index{generator.seqan2};
```

Comment:
Why do you store this in a member?
```suggestion
            typename sequence_generator<index_tag_t, rng_t>::index_t index{generator.make_seqan2_sequence()};
```